staging/src/k8s.io/client-go/tools/cache/shared_informer.go

```diff
@@ -150,6 +152,15 @@ type SharedInformer interface {
 	// because the implementation takes time to do work and there may
 	// be competing load and scheduling noise.
 	AddEventHandlerWithResyncPeriod(handler ResourceEventHandler, resyncPeriod time.Duration)
+	// RemoveEventHandler removes a formerly added event handler.
+	// It is possible to remove only event handlers that are (go) comparable,
```

Comment:
This is a strange restriction for a method signature that takes an interface. It's also disconcerting since most ResourceEventHandler calls use non-pointer ResourceEventHandlerFuncs, and the error when trying to remove is a runtime error, not a compile error.

Either returning an handle that could be passed to RemoveEventHandler to remove the handler (regardless of whether the original handler was comparable), or passing in something that could be used to explicitly remove the handler (stopChannel that can be closed, etc) might be better.

Reply:
The I would prefer to return a handle. But anyway this is much more invasive change, because the internal 
structure has to be changed. I'll try to prepare a proposal the next days.

Reply:
May be we could wrap the uncomparable handler into a handle implementing the ResourceEventHandler interface. So, if the handler is comparable, we use the handler as handle, and if not it is wrapped. 
The Add methods hist return the resulting effective handler, that can always be removed.

Or do you pefer a more explicit Handle interface. This could avoid the runtime exception at all.

Reply:
There are already other implementations for a `SharedIndexInformer` interface wrapping a base informer.
Those will break if I change the signature of an existing method.
Is it ok to change the signature (and adapting the implementations in the kubernetes project)?

Reply:
@liggitt I've added a commit introducing an explicit handle. I could stick to the handler interface internally. Therefore the change is less invasive than expected. But the signature change broke a wrapper of a shared informer, Therefore I had to change something in the replication controller, also.